WalletWasabi.Fluent/Views/AddWallet/Create/ConfirmRecoveryWordsView.axaml

```diff
@@ -12,29 +12,95 @@
-      <ItemsControl Items="{Binding ConfirmationWords}">
-        <i:Interaction.Behaviors>
-          <behaviors:FocusFirstTextBoxInItemsControlBehavior />
-        </i:Interaction.Behaviors>
-        <ItemsControl.ItemTemplate>
-          <DataTemplate>
-            <DockPanel>
-              <Label Content="{Binding Index, StringFormat=Recovery word {0}, Mode=OneWay}" DockPanel.Dock="Top" />
-              <TextBox Watermark="Type in the word" Classes="hasCheckMarkBoolean" validation:CheckMarkStatus.IsEnabled="{Binding IsConfirmed}" Text="{Binding Input}" IsEnabled="{Binding !IsConfirmed}" DockPanel.Dock="Top">
-                <i:Interaction.Behaviors>
-                  <behaviors:FocusNextItemBehavior IsFocused="{Binding !IsConfirmed}" />
-                </i:Interaction.Behaviors>
-              </TextBox>
-            </DockPanel>
+                 EnableNext="True"
+                 NextContent="Continue"
+                 EnableSkip="{Binding IsSkipEnabled}"
+                 IsBusy="{Binding IsBusy}"
+                 ScrollViewer.VerticalScrollBarVisibility="Disabled">
+    <Grid x:Name="Layout">
+      <Grid.RowDefinitions>
+        <RowDefinition MinHeight="70" Height="*" />
+        <RowDefinition Height="16" />
+        <RowDefinition Height="Auto" />
+        <RowDefinition Height="*" />
+        <RowDefinition Height="20" />
+      </Grid.RowDefinitions>
+
+      <c:RecoveryWordsControl Items="{Binding AvailableWords}">
+        <c:RecoveryWordsControl.Styles>
+          <Style Selector="StackPanel">
+            <Setter Property="HorizontalAlignment" Value="Center" />
+          </Style>
+          <Style Selector="ItemsControl">
+            <Setter Property="ItemsPanel">
+              <ItemsPanelTemplate>
+                <StackPanel Orientation="Horizontal" />
+              </ItemsPanelTemplate>
+            </Setter>
+          </Style>
+        </c:RecoveryWordsControl.Styles>
+        <c:RecoveryWordsControl.ItemTemplate>
+          <DataTemplate x:DataType="create:RecoveryWordViewModel">
+            <CheckBox IsChecked="{Binding IsSelected}"
+                      IsEnabled="{Binding IsEnabled}">
+              <CheckBox.Styles>
+                <Style Selector="CheckBox[IsChecked=True] /template/ c|TagControl">
+                  <Setter Property="Foreground" Value="{DynamicResource SystemAccentColor}" />
+                  <Setter Property="BorderBrush" Value="{DynamicResource SystemAccentColor}" />
+                </Style>
+                <Style Selector="CheckBox[IsEnabled=False] /template/ c|TagControl">
+                  <Setter Property="Foreground" Value="{DynamicResource ButtonForegroundDisabled}" />
+                  <Setter Property="BorderBrush" Value="{DynamicResource ButtonForegroundDisabled}" />
+                </Style>
+              </CheckBox.Styles>
+              <CheckBox.Template>
+                <ControlTemplate>
```

Comment:
Dont need a template here.

Reply:
Not using a template makes the UI look bad due to existing ToggleButton styles.